lib/Parse/Lexer.cpp

```diff
@@ -2302,42 +2302,34 @@ void Lexer::lexTrivia(syntax::TriviaList &Pieces, bool IsForTrailingTrivia) {
     if (IsForTrailingTrivia)
       break;
     NextToken.setAtStartOfLine(true);
-    LLVM_FALLTHROUGH;
-  case ' ':
-  case '\t':
-  case '\v':
-  case '\f': {
-    auto Char = CurPtr[-1];
-    // Consume consective same characters.
-    while (*CurPtr == Char)
-      ++CurPtr;
-
-    auto Length = CurPtr - TriviaStart;
-    switch (Char) {
-    case ' ':
-      Pieces.push_back(TriviaPiece::spaces(Length));
-      break;
+    switch (CurPtr[-1]) {
```

Comment:
We can assume that most people only use `\n`.  So, I think, probably
```cpp
case '\n'
  if (IsForTrailingTrivia)
    break;
  NextToken.setAtStartOfLine(true);
  Pieces.appendOrSquash(TriviaPiece::newlines(1));
case '\r':
  if (IsForTrailingTrivia)
    break;
  NextToken.setAtStartOfLine(true);
  if (*CurPtr == '\n') {
    Pieces.appendOrSquash(TriviaPiece::carriageReturnLineFeeds(1));
```
is slightly more efficient.

Reply:
I see. I change so.